### ISSUE TITLE
This API is a work in progress and will likely be changing frequently. 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#2794]
Given initial gradient-node outputs 'y_grad_node_outputs' (which compute the
symbolic partial derivatives of some loss function 'L' w.r.t the node outputs
'y_node_outputs'), adds gradient nodes to 'graph' that compute the symbolic
partial derivatives of 'L' w.r.t the node outputs 'x_node_outputs'.

REQUIRES: Each node in 'x_node_outputs' to be unique, and so to have a single
output (this restriction will be removed in a subsequent change).

----
*opened via [imdone.io](https://imdone.io) from a code comment on [f4ade48bc73257724e0069b3e2cdf41e96fcc32c](https://github.com/imdone/tensorflow/commit/f4ade48bc73257724e0069b3e2cdf41e96fcc32c) by A. Unique TensorFlower nobody@tensorflow.org*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/core/graph/gradients.h#L36-L49
